src/desktop/apps/experimental-app-shell/utils/__tests__/userRequiredMiddleware.jest.ts

```diff
@@ -18,30 +18,86 @@ describe("orderMiddleware", () => {
     expect(next).toHaveBeenCalled()
   })
 
-  it("performs redirect if CURRENT_USER=undefined", () => {
-    const spy = jest.fn()
-    const req = {
-      path: "/orders/",
-      url: "/orders/",
-      originalUrl: "/orders/",
-      query: {
-        "accepted-conditions": false,
-      },
-    }
-    const res = {
-      redirect: spy,
-      locals: {
-        sd: {
-          CURRENT_USER: null,
+  describe("user-required routes perform redirect if CURRENT_USER=undefined", () => {
+    it("performs redirect on an /auction-registration route", () => {
+      const spy = jest.fn()
+      const req = {
+        path: "/auction-registration/",
+        url: "/auction-registration/",
+        originalUrl: "/auction-registration/",
+        query: {
+          "accepted-conditions": false,
         },
-      },
-    }
+      }
+      const res = {
+        redirect: spy,
+        locals: {
+          sd: {
+            CURRENT_USER: null,
+          },
+        },
+      }
 
-    const next = jest.fn()
-    userRequiredMiddleware(req, res, next)
-    expect(spy).toHaveBeenCalledWith(
-      `/login?redirectTo=${encodeURIComponent(req.originalUrl)}`
-    )
-    expect(next).not.toHaveBeenCalled()
+      const next = jest.fn()
+      userRequiredMiddleware(req, res, next)
+      expect(spy).toHaveBeenCalledWith(
+        `/login?redirectTo=${encodeURIComponent(req.originalUrl)}`
+      )
+      expect(next).not.toHaveBeenCalled()
+    })
+
+    it("performs redirect on a /orders route", () => {
+      const spy = jest.fn()
+      const req = {
+        path: "/orders/",
+        url: "/orders/",
+        originalUrl: "/orders/",
+        query: {
+          "accepted-conditions": false,
+        },
+      }
+      const res = {
+        redirect: spy,
+        locals: {
+          sd: {
+            CURRENT_USER: null,
+          },
+        },
+      }
+
+      const next = jest.fn()
+      userRequiredMiddleware(req, res, next)
+      expect(spy).toHaveBeenCalledWith(
+        `/login?redirectTo=${encodeURIComponent(req.originalUrl)}`
+      )
+      expect(next).not.toHaveBeenCalled()
+    })
+
+    it("performs redirect on an /identity-verification route", () => {
+      const spy = jest.fn()
+      const req = {
+        path: "/identity-verification/123",
+        url: "/identity-verification/123",
+        originalUrl: "/identity-verification/123",
+        query: {
+          "accepted-conditions": false,
+        },
+      }
+      const res = {
+        redirect: spy,
+        locals: {
+          sd: {
+            CURRENT_USER: null,
+          },
+        },
+      }
+
+      const next = jest.fn()
+      userRequiredMiddleware(req, res, next)
+      expect(spy).toHaveBeenCalledWith(
+        `/login?redirectTo=${encodeURIComponent(req.originalUrl)}`
+      )
+      expect(next).not.toHaveBeenCalled()
+    })
```

Comment:
Can we add a test in this file that ensure's we're matching each of the lines [here](https://github.com/artsy/force/blob/92396459569d0aa8fb71f6b25ddcaca64e227cc1/src/desktop/apps/experimental-app-shell/utils/userRequiredMiddleware.tsx#L3-L8)

Also, mind moving `userRequiredMiddleware.tsx` to a `middleware` folder at the same level as `utils` and `apps`? It's a folder that will likely grow.

Reply:
done :+1: